resources/lib/datafunctions.py

```diff
@@ -474,48 +474,84 @@ def _get_additionalproperties( self, profileDir ):
-                                    defaultProperties.append( [ "mainmenu", elem.attrib.get( "labelID" ), "backgroundName", backgroundName, elem.attrib.get( 'defaultID' ) ] )
+                                    defaultProperties.append( [ "mainmenu", labelID, "backgroundName", backgroundName, defaultID ] )
                             
                             if elemSearch[0] == "widget":
                                 # Get and set widget type and name
                                 widgetDetails = self._getWidgetNameAndType( elem.text )
                                 if widgetDetails is not None:
-                                    defaultProperties.append( [ elem.attrib.get( "group" ), elem.attrib.get( "labelID" ), "widgetName", widgetDetails[0], elem.attrib.get( 'defaultID' ) ] )
-                                    if widgetDetails[1] is not None:
-                                        defaultProperties.append( [ elem.attrib.get( "group" ), elem.attrib.get( "labelID" ), "widgetType", widgetDetails[1], elem.attrib.get( 'defaultID' ) ] )                
+                                    defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetName", widgetDetails[ "name" ], defaultID ] )
+                                    if "type" in widgetDetails:
+                                        defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetType", widgetDetails[ "type" ], defaultID ] )
+                                    if "path" in widgetDetails:
+                                        defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetPath", widgetDetails[ "path" ], defaultID ] )
+                                    if "target" in widgetDetails:
+                                        defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetTarget", widgetDetails[ "target" ], defaultID ] )
+
+                            if elemSearch[ 0 ] == "widget:node":
+                                # Set all widget properties from the default
+                                if "label" in elem.attrib:
+                                    defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetName", elem.attrib.get( "label" ), defaultID ] )
+                                if "type" in elem.attrib:
+                                    defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetType", elem.attrib.get( "type" ), defaultID ] )
+                                if "path" in elem.attrib:
+                                    defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetPath", elem.attrib.get( "path" ), defaultID ] )
+                                if "target" in elem.attrib:
+                                    defaultProperties.append( [ elem.attrib.get( "group" ), labelID, "widgetTarget", elem.attrib.get( "target" ), defaultID ] )
                                         
```

Comment:
I always prefer something like this for all kind of repititions:
```python
dct = {"label": "widgetName",
	   ....}
for key, value in dct.items():
	if key in elem.attrib:
		defaultProperties.append(....)
```
(either using dict or using a list of tuples)